Client/Frontend/Login Management/LoginListViewModel.swift

```diff
@@ -0,0 +1,113 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Storage
+import Shared
+
+// MARK: - Main View Model
+// Login List View Model
+final class LoginListViewModel {
+
+    let profile: Profile
+    fileprivate var activeLoginQuery: Deferred<Maybe<[LoginRecord]>>? 
+    var isDuringSearchControllerDismiss = false
+    var count = 0
+    weak var searchController: UISearchController?
+    var titles = [Character]()
+    var loginRecordSections = [Character: [LoginRecord]]() {
+        didSet {
+            assert(Thread.isMainThread)
+            delegate?.loginSectionsDidUpdate()
+        }
+    }
+    fileprivate let helper = LoginListDataSourceHelper()
```

Comment:
Should most of these be private set or should other classes/structs be able to change them?